### PR TITLE
e2e: vanilla chain <-> contract interaction 

### LINKED ISSUE #6: e2e test setup with babylon contracts
- Make use of admin commands in Babylon contracts

### PR DESCRIPTION
**Pull request by SebastianElvis**
Fixes #6 

This PR provides the vanilla setup for e2e test, including 

- a test suite spinning up an (idle) provider chain, a consumer chain, and babylon contracts on the consumer chain
- submitting a FP to BTC staking contract via admin commands
- bumping cosmos-sdk dependency to v0.50.6 to make setup compatible with latest Babylon

Next steps would be sending custom msgs to Babylon contracts upon BeginBlock/EndBlock.

To test, run `make test-e2e`

### CODE REVIEW

File: x/babylon/keeper/handler_plugin.go [hunk 1]
```diff
@@ -46,28 +47,28 @@ func defaultMaxCapAuthorizator(k *Keeper) AuthSourceFn {
 }
 
 // DispatchMsg handle contract message of type Custom in the babylon namespace
-func (h CustomMsgHandler) DispatchMsg(ctx sdk.Context, contractAddr sdk.AccAddress, _ string, msg wasmvmtypes.CosmosMsg) ([]sdk.Event, [][]byte, error) {
+func (h CustomMsgHandler) DispatchMsg(ctx sdk.Context, contractAddr sdk.AccAddress, _ string, msg wasmvmtypes.CosmosMsg) ([]sdk.Event, [][]byte, [][]*codectypes.Any, error) {
```
Comment (maurolacy) [inline x/babylon/keeper/handler_plugin.go:50]: Not sure / clear why this change is needed.

File: tests/e2e/test_client.go [hunk 2]
```diff
@@ -47,97 +46,113 @@ func (q QueryResponse) Array(key string) []QueryResponse {
...
-	rsp, err := p.Exec(contract, payload, funds...)
-	require.NoError(p.t, err)
-	return rsp
-}
-
-func (p TestProviderClient) Exec(contract sdk.AccAddress, payload string, funds ...sdk.Coin) (*abci.ExecTxResult, error) {
-	rsp, err := p.chain.SendMsgs(&wasmtypes.MsgExecuteContract{
-		Sender:   p.chain.SenderAccount.GetAddress().String(),
+func (p *TestProviderClient) Exec(contract sdk.AccAddress, payload []byte, funds ...sdk.Coin) (*abci.ExecTxResult, error) {
+	rsp, err := p.Chain.SendMsgs(&wasmtypes.MsgExecuteContract{
```
Comment (gusin13) [inline tests/e2e/test_client.go:70]: can you explain how `SendMsgs` work, i briefly checked and it seems to be some helper func in ibctesting module which I didn't understand. I thought we need to do something like this `wasmd tx wasm instantiate <args>`

**Review by gusin13 — APPROVED** [review]
nice work, lgtm ✅

File: tests/e2e/test_client.go (hunk 2 above)
Comment (gusin13) [reply tests/e2e/test_client.go:70]: also if you could explain what is `p.Chain` here, does this start a wasmd node in background?

File: go.work.sum [hunk 3]
```diff
@@ -34,6 +34,9 @@ cloud.google.com/go/compute v1.20.1/go.mod h1:4tCnrn48xsqlwSAiLf1HXMQk8CONslYbdi
 cloud.google.com/go/compute v1.23.0/go.mod h1:4tCnrn48xsqlwSAiLf1HXMQk8CONslYbdiEZc9FEIbM=
```
Comment (gusin13) [inline go.work.sum:34]: what is a `go.work.sum` file? looks like `go.mod` ?

File: tests/e2e/test_client.go (hunk 2 above)
Comment (SebastianElvis) [reply tests/e2e/test_client.go:70]: So all of these come from the `ibctesting` [module](https://github.com/CosmWasm/wasmd/tree/main/x/wasm/ibctesting) that allows to write unit tests about IBC protocols. The `p.Chain` is a Cosmos SDK chain defined in the [TestChain](https://github.com/CosmWasm/wasmd/tree/main/x/wasm/ibctesting) struct. You can see it includes a fully fledged `App` and various metadata. `SendMsgs` allows us to send arbitrary message to this chain.

In L70, we send a `MsgExecuteContract` tmessage o the chain. `MsgExecuteContract` is used for submitting a certain message to a certain smart contract. For Babylon SDK, we will be using this command to send a message to BTC staking contract upon each `BeginBlock`, such that BTC staking contract can update the voting power table, tally non-finalised blocks, etc..

File: go.work.sum (hunk 3 above)
Comment (SebastianElvis) [reply go.work.sum:34]: This is a file used for [Go workspace](https://go.dev/doc/tutorial/workspaces). A go workspace allows us to maintain multiple Go modules in a single repository. You can see in `go.work` we have 

```
go 1.21

use (
	demo
	tests/e2e
	x
)
```

This defines that this Go workspace has these three modules. In each directory, you can see there are `go.mod` and `go.sum`.

In Babylon SDK, 

- `demo` contains a demo Cosmos SDK chain that installs wasm and babylon-sdk module.
- `test/e2e` contains e2e tests for Babylon sdk
- `x` contains the Babylon SDK module that will be installed to the consumers

**Review by maurolacy — COMMENTED** [review]
Looks good, good work.

File: x/babylon/keeper/handler_plugin.go (hunk 1 above)
Comment (SebastianElvis) [reply x/babylon/keeper/handler_plugin.go:50]: This interface is changed in wasmd v0.51.0
